pytest_mh/_private/plugin.py

```diff
@@ -562,6 +567,22 @@ def _clone_function(self, name: str, f: pytest.Function) -> pytest.Function:
             originalname=f.originalname,
         )
 
+    def _preferred_topology(self, item: pytest.Item):
+        if self.mh_ignore_preferred_topology:
+            self.logger.info(" --mh-ignore-preferred-topology set: Ignoring preferred topology marker!")
+            return
+
+        if len(self.mh_topology) > 0:
+            self.logger.info(" --mh-topology set: Ignoring preferred topology marker!")
+            return
+
+        for i in item.iter_markers("preferred_topology"):
+            self.preferred_topology = i.args[0] if isinstance(i.args[0], str) else i.args[0].value.name
```

Comment:
This can be local variable only.

Reply:
Also be more thorough when checking the type to make sure only allowed types were passed.